src/pages/profile/external-identities/FrameworkServiceLink.ts

```diff
@@ -0,0 +1,47 @@
+import { ClickableProps } from '@terra-ui-packages/components';
+import { Fragment, useState } from 'react';
+import { h } from 'react-hyperscript-helpers';
+import { ButtonPrimary, Link } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { Ajax } from 'src/libs/ajax';
+import { withErrorReporting } from 'src/libs/error';
+import { useOnMount } from 'src/libs/react-utils';
+import * as Utils from 'src/libs/utils';
+
+interface FrameworkServiceLinkProps extends ClickableProps {
+  linkText: string;
+  provider: string;
+  redirectUrl: string;
+  button?: boolean;
+}
```

Comment:
Isn't this "provider" actually a provider key? That was confusing to me, since they had the same name when converting `User.js` -> `User.ts`. Maybe in the cases where we are referring to the provider name we could call it `providerName` or `providerKey` to distinguish it from the provider object?